lib/Sema/CSDiag.cpp

```diff
@@ -1213,10 +1211,23 @@ CalleeCandidateInfo::evaluateCloseness(DeclContext *dc, Type candArgListType,
   // close matches are prioritized against obviously wrong ones.
   if (mismatchingArgs == 1) {
     CandidateCloseness closeness;
-    if (singleArchetype.isNull()) {
+    if (allGenericSubstitutions.size() == 0) {
       closeness = mismatchesAreNearMisses ? CC_OneArgumentNearMismatch
                                           : CC_OneArgumentMismatch;
     } else {
+      // If the failure is that different occurrences of the same generic have
+      // different concrete types, substitute in all the concrete types we've found
+      // into the failureInfo to improve diagnosis.
+      if (genericWithDifferingConcreteTypes) {
+        auto newType = failureInfo.parameterType.transform([&](Type type) -> Type {
```

Comment:
Unfortunately, no, because the subst() transform puts in SubstitutedTypes, and those emit different descriptions with "aka" in the diagnosis. E.g. "cannot convert value of type 'Float' to expected argument type '(Int)' (aka 'Int')"